beetmoverscript/task.py

```diff
@@ -188,21 +188,41 @@ def update_props(context, props, platform_mapping):
     return props
 
 
-def get_updated_buildhub_artifact(path, installer_path, context, manifest, locale):
+def get_updated_buildhub_artifact(path, installer_path, context, locale, manifest=None, artifact_map=None):
     """
     Read the file into a dict, alter the fields below, and return the updated dict
     buildhub.json fields that should be changed: download.size, download.date, download.url
     """
     contents = utils.load_json(path)
     installer_name = os.path.basename(installer_path)
-    dest = manifest['mapping'][locale][installer_name]['destinations']
+    if artifact_map:
+        task_id = get_taskId_from_full_path(path)
+        dest = artifact_map[task_id][installer_name]['destinations']
+    else:
+        dest = manifest['mapping'][locale][installer_name]['destinations']
     url_prefix = context.config["bucket_config"][context.bucket]["url_prefix"]
     # assume url_prefix is ASCII safe
-    url = urllib.parse.quote(urllib.parse.urljoin(manifest["s3_bucket_path"], dest[0]))
+    url = urllib.parse.quote(urllib.parse.urljoin(url_prefix, dest[0]))
 
     # Update fields
     contents['download']['size'] = utils.get_size(installer_path)
     contents['download']['date'] = str(arrow.utcnow())
     contents['download']['url'] = urllib.parse.urljoin(url_prefix, url)
 
     return contents
+
+
+def get_taskId_from_full_path(full_path_artifact):
+    """ Temporary fix: Extract the taskId from a full path artifact
```

Comment:
Good point. We might find a better solution here once we can test things end-to-end.

Reply:
For proper attribution, it was Lisa's comment, I just left it there because I agreed!